libs/ledger/src/protocols/main_chain_rpc_service.cpp

```diff
@@ -393,11 +378,21 @@ MainChainRpcService::State MainChainRpcService::OnWaitForHeaviestChain()
             next_state = State::SYNCHRONISING;  // we have reached the tip
           }
         }
+        else
+        {
+          FETCH_LOG_WARN(LOGGING_NAME,
+                         "Received empty block response when forward syncing the chain!");
+          state_machine_->Delay(std::chrono::seconds{8});
+          next_state = GetInitialState(mode_);
+        }
       }
       else
       {
         FETCH_LOG_INFO(LOGGING_NAME, "Heaviest chain request to: ", ToBase64(current_peer_address_),
                        " failed. Reason: ", service::ToString(status));
+
+        state_machine_->Delay(std::chrono::seconds{1});
+        next_state = GetInitialState(mode_);
```

Comment:
If this condition happened, it would spin in this state forever printing the log message